test/metabase/api/setup_test.clj

```diff
@@ -272,6 +274,39 @@
         (with-setup {:database {:engine "h2", :name db-name}}
           (is (db/exists? Database :name db-name)))))))
 
+(deftest create-superuser-only-once-test
+  (testing "POST /api/setup"
+    (testing "Check that we cannot create a new superuser via setup-token when a user exists"
+      (let [token (setup/create-token!)
+            body  {:token token
+                   :prefs {:site_locale "es_MX"
+                           :site_name   (mt/random-name)}
+                   :user  {:first_name (mt/random-name)
+                           :last_name  (mt/random-name)
+                           :email      (mt/random-email)
+                           :password   "p@ssword1"}}]
+
+        (is (setup/has-user-setup))
```

Comment:
If this test is run with `clj -X:dev:test :only metabase.api.setup-test/create-superuser-only-once-test` i suspect this line will fail? It seems we will need to ensure that the user information is completely mocked in this test

Reply:
There are actually 4 users setup in the test database, so has-user-setup is 'set'. I was working on a function wrapper that would save, and delete them, run the test, and reinstate them, but cannot get it to work in a way that doesn't pollute the test db and break _other_ tests.

Reply:
That's fair. I'm just not positive that it is always the case that there are four users in the database under scenarios where you run the tests for an individual namespace from the cli.

Reply:
Gotcha, I'll remove that line and add a test that takes that into account.